package/upgrade/migrations/upgrade_manifests/v1.4.0/pre-hook.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash -ex
+
+patch_harvester_vm_dashboard()
+{
+    echo "Patch configmap/harvester-vm-dashboard..."
+    local patch_json=$( \
+        kubectl get -n cattle-dashboards configmap harvester-vm-dashboard -o json | \
+            jq '.data."harvester_vm_dashboard.json" | fromjson | .panels |=
+                map(
+                    if .title == "CPU Usage" then
+                        .yaxes |= map(
+                            if .format == "percentunit" then
+                                .format = "percent"
+                            else
+                                .
+                            end
+                    )
+                    else
+                        .
+                    end
+                )' |
+        jq -r .
+    )
+
+    if [ -z "$patch_json" ]; then
+        echo "Something goes wrong, the patch json is empty... skip this patch"
+        return 1
+    fi
+
+    patch_json=$(echo '{"data":{"harvester_vm_dashboard.json": ""}}' | jq --arg arg "$patch_json" '.data."harvester_vm_dashboard.json"=$arg')
+    kubectl patch -n cattle-dashboards configmap harvester-vm-dashboard --type merge -p "$patch_json"
+}
+
+patch_harvester_vm_detail_dashboard()
+{
+    echo "Patch configmap/harvester-vm-detail-dashboard..."
+    local patch_json=$( \
+        kubectl get -n cattle-dashboards configmap harvester-vm-detail-dashboard -o json | \
+            jq '.data."harvester_vm_info_details.json" | fromjson | .panels |=
+                map(
+                    if .title == "CPU Usage" then
+                        .fieldConfig.defaults.unit = "percent"
+                    else
+                        .
+                    end
+                )'
+    )
+
+    if [ -z "$patch_json" ]; then
+        echo "Something goes wrong, the patch json is empty... skip this patch"
+        return 1
+    fi
+
+    patch_json=$(echo '{"data":{"harvester_vm_info_details.json": ""}}' | jq --arg arg "$patch_json" '.data."harvester_vm_info_details.json"=$arg')
+    kubectl patch -n cattle-dashboards configmap harvester-vm-detail-dashboard --type merge -p "$patch_json"
+}
+
+# Fix incorrect unit in CPU Usage dashboard https://github.com/harvester/harvester/issues/7086
+patch_harvester_vm_dashboard
```

Comment:
Are there any side effects if this script is reentered again?

Reply:
I believe no, the script only replace unit in cpu usage dashboard from percentunit to percent. If there is no matches (e.g. no cpu usage dashboard, or the unit is already percent), it should remain the json as it is.

Below is my test that execute pre-hook.sh multiple times when two cpu usage dashboards unit are already `percent` (the expected unit). And everything is ok.
```
harvester-node-0:~ # sh pre-hook.sh
Patch configmap/harvester-vm-dashboard...
configmap/harvester-vm-dashboard patched (no change)
Patch configmap/harvester-vm-detail-dashboard...
configmap/harvester-vm-detail-dashboard patched (no change)
harvester-node-0:~ # sh pre-hook.sh
Patch configmap/harvester-vm-dashboard...
configmap/harvester-vm-dashboard patched (no change)
Patch configmap/harvester-vm-detail-dashboard...
configmap/harvester-vm-detail-dashboard patched (no change)
```